### PR TITLE
Feature/1.1/#31 startup resource credits fix

### LINKED ISSUE #31: Startup resource credit calculation is wrong and needs to be adapted to Lighthive
Need to build

https://lighthive.readthedocs.io/en/latest/helpers.html#resourcecredits-helper

Into startup for server mode.

### PR DESCRIPTION
**Pull request by brianoflondon**
Closes #31 with startup checks and extends test time on simulcast operation.

Github says this will only close the issue if the pull request is to the default branch.

### CODE REVIEW

**Review by brianoflondon — COMMENTED** [review]
I think I've done the job here. Startup is streamlined and only posts one custom_json while resource credit usage is calculated by the Lighthive library.

Also added a proper test for the startup code (because it wasn't covered by tests before).

File: src/podping_hivewriter/podping_hivewriter.py [hunk 1]
```diff
@@ -161,32 +161,31 @@ async def test_hive_resources(self):
                 "uuid": str(uuid.uuid4()),
                 "hive": str(self.lighthive_client.current_node),
             }
-
-            await self.send_notification(custom_json, STARTUP_OPERATION_ID)
-            settings = await self.settings_manager.get_settings()
-            logging.info(
-                f"Testing Account Resource Credits {settings.hive_operation_period}s"
+            op, size_of_json = await self.construct_operation(
+                custom_json, STARTUP_OPERATION_ID
```
Comment (agates) [inline src/podping_hivewriter/podping_hivewriter.py]: Note sure why this is working but this is an invalid type... you have construct_operation() requiring "HiveOperationId" but giving it a string.

Also, STARTUP_OPERATION_ID is just set to "_startup".  Is this what you intended to set as the operation_id?

File: src/podping_hivewriter/podping_hivewriter.py [hunk 2]
```diff
@@ -161,32 +161,31 @@ async def test_hive_resources(self):
...
             custom_json["v"] = podping_hivewriter_version
-            custom_json["capacity"] = f"{capacity:.1f}"
+            custom_json["capacity"] = f"{capacity:,.0f}"
             custom_json["message"] = "Podping startup complete"
             custom_json["hive"] = str(self.lighthive_client.current_node)
 
-            await self.send_notification(custom_json, STARTUP_OPERATION_ID)
+            hive_operation_id = self.operation_id + STARTUP_OPERATION_ID
+
+            await self.send_notification(custom_json, hive_operation_id)
```
Comment (agates) [inline src/podping_hivewriter/podping_hivewriter.py]: Similar issue to line 164 above... str is an invalid type for send_notification.  Maybe we could change the signature to `Union[HiveOperationId, str]`?

File: src/podping_hivewriter/constants.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 LIVETEST_OPERATION_ID = "pplt"
 PODPING_OPERATION_ID = "pp"
-STARTUP_OPERATION_ID = "ppst"
+STARTUP_OPERATION_ID = "_startup"
```
Comment (agates) [inline src/podping_hivewriter/constants.py:3]: If this is intended to be a suffix to either `LIVETEST_OPERATION_ID` or `PODPING_OPERATION_ID` we should probably indicate as such.

**Review by agates — CHANGES_REQUESTED** [review]
Couple of small things but makes sense overall.

File: src/podping_hivewriter/podping_hivewriter.py (hunk 1 above)
Comment (brianoflondon) [reply src/podping_hivewriter/podping_hivewriter.py]: What I'd like is for all pings to start with pp_ or pplt_ for podping and podping livetest when we're testing.

What I really don't understand is how the Operation ID works with the GPRC or whatever that google compiler is.

So whatever the correct way to add _startup for this (now single) custom_json... that's what we should do.

File: src/podping_hivewriter/constants.py (hunk 3 above)
Comment (brianoflondon) [reply src/podping_hivewriter/constants.py:3]: Better comment elsewhere.

**Comment by brianoflondon** [comment]
I'm pretty sure the whole way startup is checked is now much more reliable and solid.

Resource credit calc is also as good as it can be.

**Review by agates — APPROVED** [review]

**Pull request merged**